### PR TITLE
NickAkhmetov/HMP-159 Upgrade vitessce-python dependency

### PR DESCRIPTION
**Pull request by NickAkhmetov**
This PR updates the `vitessce-python` dependency from 1.0.9 to 3.0.5; this was done as part of troubleshooting HMP-296. There are still failing tests etc, which will be updated accordingly once the core issue in 296 is resolved.

### CODE REVIEW

**Comment by NickAkhmetov** [comment]
Current state:
- All the assays on the portal which have vitessce hints display identical to prod
- Potential issue: I can't get the `obsLabelsType` coordination that's supposed to take the place of the `factors` to work as expected right now. When the related code is uncommented and the duplicate obs sets are removed, the tooltips show up with blank values for all marker genes, regardless of whether the coordination value is prefixed with `obs`. Continuing to investigate this a bit further.

Without `obs/`:
image

With `obs/`:
image


<details><summary>Sample vitessce config with the marker genes issue</summary>
<p>

```
{
  "coordinationSpace": {
    "dataset": {
      "A": "A"
    },
    "embeddingType": {
      "A": "UMAP"
    },
    "obsLabelsType": {
      "A": "marker_gene_0",
      "B": "marker_gene_1",
      "C": "marker_gene_2",
      "D": "marker_gene_3",
      "E": "marker_gene_4"
    }
  },
  "datasets": [
    {
      "files": [
        {
          "fileType": "anndata.zarr",
          "options": {
            "featureLabels": {
              "path": "var/hugo_symbol"
            },
            "obsEmbedding": [
              {
                "dims": [
                  0,
                  1
                ],
                "embeddingType": "UMAP",
                "path": "obsm/X_umap"
              }
            ],
            "obsFeatureMatrix": {
              "initialFeatureFilterPath": "var/marker_genes_for_heatmap",
              "path": "X"
            },
            "obsSets": [
              {
                "name": "Leiden",
                "path": "obs/leiden"
              }
            ]
          },
          "url": "https://assets.hubmapconsortium.org/6c57274e7a40413dc042ec32442a228b/hubmap_ui/anndata-zarr/secondary_analysis.zarr"
        }
      ],
      "name": "6c57274e7a40413dc042ec32442a228b",
      "uid": "A"
    }
  ],
  "description": "",
  "initStrategy": "auto",
  "layout": [
    {
      "component": "scatterplot",
      "coordinationScopes": {
        "dataset": "A",
        "embeddingType": "A",
        "obsLabelsType": [
          "A",
          "B",
          "C",
          "D",
          "E"
        ]
      },
      "h": 6,
      "w": 9,
      "x": 0,
      "y": 0
    },
    {
      "component": "obsSets",
      "coordinationScopes": {
        "dataset": "A",
        "obsLabelsType": [
          "A",
          "B",
          "C",
          "D",
          "E"
        ]
      },
      "h": 3,
      "w": 3,
      "x": 9,
      "y": 0
    },
    {
      "component": "featureList",
      "coordinationScopes": {
        "dataset": "A",
        "obsLabelsType": [
          "A",
          "B",
          "C",
          "D",
          "E"
        ]
      },
      "h": 3,
      "w": 3,
      "x": 9,
      "y": 4
    },
    {
      "component": "obsSetFeatureValueDistribution",
      "coordinationScopes": {
        "dataset": "A",
        "obsLabelsType": [
          "A",
          "B",
          "C",
          "D",
          "E"
        ]
      },
      "h": 4,
      "w": 5,
      "x": 7,
      "y": 6
    },
    {
      "component": "heatmap",
      "coordinationScopes": {
        "dataset": "A",
        "obsLabelsType": [
          "A",
          "B",
          "C",
          "D",
          "E"
        ]
      },
      "h": 4,
      "w": 7,
      "x": 0,
      "y": 6
    }
  ],
  "name": "6c57274e7a40413dc042ec32442a228b",
  "version": "1.0.15"
}
```

</p>
</details>

**Comment by NickAkhmetov** [comment]
I put together a local draft release of `vitessce-python 3.0.7` using some of my pending PR's:
- https://github.com/vitessce/vitessce-python/pull/272
- https://github.com/vitessce/vitessce-python/pull/273
- https://github.com/vitessce/vitessce-python/pull/274

I verified these changes work as expected locally; CI fails for this branch because 3.0.7 is not yet a released Vitessce version. Once those three PR's are merged and a new release is made, I will update the test fixtures again and open this PR for review.

I've also preemptively pushed a `portal-visualization` package version bump as part of this PR given the change in dependency version.

**Comment by NickAkhmetov** [comment]
- Vitessce 3.0.7 is released
- Fixtures are updated
- Tests pass
- Smoke test of portal visualizations loaded as expected

Ergo, I'm opening this for review.

File: .vscode/settings.json
Comment (NickAkhmetov) [inline .vscode/settings.json:2]: I added this .vscode config as a guide to avoid long comment lines that broke lint

File: src/portal_visualization/builders/anndata_builders.py
Comment (john-conroy) [inline src/portal_visualization/builders/anndata_builders.py:65]: Can we define `schema_version` somewhere for reuse?

**Review by john-conroy — APPROVED** [review]
Not sure if my review is valuable here, but looks good to me!

File: src/portal_visualization/builders/anndata_builders.py
Comment (NickAkhmetov) [reply src/portal_visualization/builders/anndata_builders.py:65]: Great call! I've added this as a private field in the `BaseBuilder` which allows for overrides to be done (if ever needed) by passing `schema_version` to the args of any builder.

File: src/portal_visualization/builders/scatterplot_builders.py
Comment (NickAkhmetov) [reply src/portal_visualization/builders/scatterplot_builders.py]: I added a `create_coordination_values` util which:
- Returns `{ "obsType": "cell" } by default (most common case)
- Can have a different `obs_type` specified by passing an arg (first positional arg or named `obs_type`)
- Can have other keys specified as named args, e.g. `embeddingType="UMAP"`

**Comment by NickAkhmetov** [comment]
After resolving John's comments above and getting verbal approval from Mark via slack, I'm merging these changes in 🙌

**Pull request merged**
